cuqi/implicitprior/_restorator.py

```diff
@@ -0,0 +1,220 @@
+#%%
+from abc import ABC, abstractmethod
+from cuqi.distribution import Distribution
+import numpy as np
+    
+class RestorationPrior(Distribution):
+    """    
+    This class defines an implicit distribution associated with a restoration operator
+    (eg denoiser). They are several works relating restorations operators with
+    priors, see 
+        -Laumont et al. https://arxiv.org/abs/2103.04715
+        -Hu et al. https://openreview.net/pdf?id=x7d1qXEn1e
+    We cannot sample from this distribution, neither compute its logpdf except in
+    some cases. It allows us to apply algorithms such as MYULA and PnPULA.
+    
+    Parameters
+    ---------- 
+    restorator : callable f(x, restoration_strength)
+        Function f that accepts input x to be restored and returns the
+        restored version of x and information about the restoration operation.
+            
+    restorator_kwargs : dictionary
+        Dictionary containing information about the restorator.
+    
+    potential : callable function, optional
+        The potential corresponds to the negative logpdf when it is accessible.
+        This function is a mapping from the parameter domain to the real set. 
+        It can be provided if the user knows how to relate it to the restorator.
+        Ex: restorator is the proximal operator of the total variation (TV), then
+        potential is the TV function. 
```

Comment:
I suggest adding explanation of what kwargs is used for here

Reply:
You are completely right. We clarified the role of restorator_kwargs providing examples. It basically contains keyword argument parameters that will be passed to the restorator f. An example could be algorithm parameters such as the number of iterations or the stopping criteria eg.